pkgs/by-name/vd/vdhcoapp/package.nix

```
@@ -0,0 +1,77 @@
+{ lib
+, fetchFromGitHub
+, buildNpmPackage
+, toml2json
+, nodejs
+, ffmpeg
+, callPackage
+, filepicker ? callPackage ./filepicker.nix {}
+, substituteAll
+, makeWrapper
+}:
+
+buildNpmPackage rec {
+  pname = "vdhcoapp";
+  version = "2.0.19";
+  srchash = "sha256-8xeZvqpRq71aShVogiwlVD3gQoPGseNOmz5E3KbsZxU=";
+
+  src = fetchFromGitHub {
+    owner = "aclap-dev";
+    repo = "vdhcoapp";
+    rev = "v${version}";
+    hash = srchash;
+  };
+
+  sourceRoot = "${src.name}/app";
+  npmDepsHash = "sha256-E032U2XZdyTER6ROkBosOTn7bweDXHl8voC3BQEz8Wg=";
+  dontNpmBuild = true;
+
+  nativeBuildInputs = [
+    toml2json
+    makeWrapper
+  ];
+
+  patches = [
+    (substituteAll {
+      src = ./ffmpeg-filepicker.patch;
+      inherit ffmpeg;
+      filepicker = lib.getExe filepicker;
+    })
+  ];
+
+  postPatch = ''
+    # Cannot use patch, setting placeholder here
+    substituteInPlace src/native-autoinstall.js \
+      --replace process.execPath "\"${placeholder "out"}/bin/vdhcoapp\""
+  '';
+
+  preBuild = ''
+    toml2json --pretty ../config.toml > src/config.json
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/opt/vdhcoapp
+
+    cp -r . "$out/opt/vdhcoapp"
+
+    makeWrapper ${nodejs}/bin/node $out/bin/vdhcoapp \
+      --add-flags $out/opt/vdhcoapp/src/main.js
+    runHook postInstall
+  '';
+
+  postInstall = ''
+    $out/bin/vdhcoapp install
+    mkdir -p $out/lib/mozilla/native-messaging-hosts
+    cp -rf /build/.mozilla/native-messaging-hosts/net.downloadhelper.coapp.json $out/lib/mozilla/native-messaging-hosts/net.downloadhelper.coapp.json
+    echo "don't forget to add vdhcoapp to programs.firefox.nativeMessagingHosts.packages in your configuration.nix"
+  '';
+
+  meta = with lib; {
+    description = "Companion application for the Video DownloadHelper browser add-on";
+    homepage = "https://www.downloadhelper.net/";
+    license = licenses.gpl2;
```

```suggestion
    license = licenses.gpl2;
    platforms = lib.platforms.linux;
```
as suggested in https://github.com/NixOS/nixpkgs/pull/300051#issuecomment-2030078101